### PR TITLE
Update NYPL DAG to use API v2

### LINKED ISSUE #2747: Change NYPL DAG to use the API v2
## Description

The v1 API for NYPL is no longer available. The DAG needs to be updated to use the v2, until that happens it remains disabled and we don't get the lastest images.

## Reproduction

1. Go to http://api.repo.nypl.org/api/v1/items/search/?q=CC_0&field=use_rtxt_s&page=1&per_page=500
2. See error message.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #2747 by @krysal

## Description

This PR updates the NYPL DAG to use the new v2 API. Fortunately almost nothing changed with the actual API interaction details! So this became a very straightforward change. The only major change was putting the tags in the right place!

I also added the option to pass arguments into the catalog's `ipython` recipe.

## Testing Instructions

1. Get the NYPL API key and add it to `AIRFLOW_VAR_API_KEY_NYPL` in the `.env`
2. Enable the NYPL workflow and let it run for however long you set your `AIRFLOW_VAR_INGESTION_LIMIT` (note that this DAG is _slow_ and will take about 3 hours to run end-to-end for ~3k results).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
Awesome that the update only took ~22 lines of code! Tested fine locally.

File: catalog/tests/dags/providers/provider_api_scripts/test_nypl.py [hunk 1]
```diff
@@ -138,12 +138,12 @@ def test_get_record_data_success():
             "date_issued": "1981",
             "genre": "Maps",
             "publisher": "New York Public Library, Local History and Genealogy Division",
-            "tags": "Census districts",
             "type_of_resource": "cartographic",
             "physical_description": "4 polyester film encapsulations, some containing 2 sheets back-to-back. "
             "Accompanying text formatted as 1 large sheet (46 x 59 cm), in one of "
             "the encapsulations.",
         },
+        "raw_tags": ["Census districts"],
```
Comment (krysal) [inline catalog/tests/dags/providers/provider_api_scripts/test_nypl.py:146]: Great to send this field to the correct place now!

File: catalog/dags/providers/provider_api_scripts/nypl.py [hunk 2]
```diff
@@ -221,6 +222,22 @@ def _get_creators(creatorinfo):
                 return info.get("namePart", {}).get("$")
         return None
 
+    @staticmethod
+    def _get_tags(mods: dict) -> list[str]:
+        subject_list = mods.get("subject", [])
+        if isinstance(subject_list, dict):
+            subject_list = [subject_list]
```
Comment (krysal) [inline catalog/dags/providers/provider_api_scripts/nypl.py:229]: Having a unit test specifically for this new method would be nice.

**Review by krysal — APPROVED** [review]
It's nice that it was a simple update!

File: catalog/dags/providers/provider_api_scripts/nypl.py (hunk 2 above)
Comment (AetherUnbound) [reply catalog/dags/providers/provider_api_scripts/nypl.py:229]: Added in [695bd7d](https://github.com/WordPress/openverse/pull/3774/commits/695bd7d162cddf0c7760edb4ff127f2c34951a36)!

**Pull request merged**